src/theory/sets/theory_sets_rels.cpp

```diff
@@ -1178,15 +1176,25 @@ void TheorySetsRels::check(Theory::Effort level)
       return true;
     } else if( hasTerm( a ) && hasTerm( b ) ){
       return d_state.areEqual(a, b);
-    } else if(a.getType().isTuple()) {
-      bool equal = true;
-      for(unsigned int i = 0; i < a.getType().getTupleLength(); i++) {
-        equal = equal && areEqual(RelsUtils::nthElementOfTuple(a, i), RelsUtils::nthElementOfTuple(b, i));
+    }
+    TypeNode atn = a.getType();
+    if (atn.isTuple())
+    {
+      size_t tlen = atn.getTupleLength();
+      for (size_t i = 0; i < tlen; i++)
+      {
+        if (!areEqual(RelsUtils::nthElementOfTuple(a, i),
+                      RelsUtils::nthElementOfTuple(b, i)))
+        {
+          return true;
```

Comment:
```suggestion
          return false;
```
I think you wanted to return false here